airbyte-commons/src/main/java/io/airbyte/commons/concurrency/GracefulShutdownHandler.java

```diff
@@ -22,19 +22,26 @@
  * SOFTWARE.
  */
 
-package io.airbyte.scheduler;
+package io.airbyte.commons.concurrency;
 
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.TimeUnit;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class SchedulerShutdownHandler extends Thread {
+public class GracefulShutdownHandler extends Thread {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(SchedulerShutdownHandler.class);
+  private static final Logger LOGGER = LoggerFactory.getLogger(GracefulShutdownHandler.class);
+  private final long terminationWaitTime;
+  private final TimeUnit terminateWaitTimeUnits;
   private final ExecutorService[] threadPools;
 
-  public SchedulerShutdownHandler(final ExecutorService... threadPools) {
+  public GracefulShutdownHandler(
+                                 long terminationWaitTime,
```

Comment:
I slightly prefer `Duration` here so the time and unit are tied together by whatever is providing it.